dap_aria_mapping/pipeline/taxonomy_pruning/preprune_topics.py

```diff
@@ -0,0 +1,174 @@
+from chatgpt_wrapper import ChatGPT, AsyncChatGPT
+from dap_aria_mapping import chatgpt_args
+
+OUTPUT_DIR = PROJECT_DIR / "outputs" / "interim" / "preprune_topics"
+OUTPUT_DIR.mkdir(parents=True, exist_ok=True)
+
+if __name__ == "__main__":
+    try:
+        logger.info("Loading preprune_topics.pkl")
+        with open(OUTPUT_DIR / "preprune_topics.pkl", "rb") as f:
+            prepruned_entities = pickle.load(f)
+    except FileNotFoundError:
+        logger.info("No preprune_topics.pkl found, creating new list.")
+        prepruned_entities = []
+
+    try:
+        logger.info("Loading entities")
+        with open(OUTPUT_DIR / "entities.pkl", "rb") as f:
+            entities = pickle.load(f)
+        logger.info(f"Loaded entities - Length: {len(entities)}")
+    except FileNotFoundError:
+        logger.info("No entities.pkl found, creating new list.")
+        entities = get_cooccurrence_taxonomy().index.tolist()
+        entities = entities[:35000]
+
+    # Prompt chatGPT with the problem
+    bot = ChatGPT(False)
+    initial_script = f"I have created a hierarchical taxonomy of topics that are relevant to the academic literature \
+        and the commercialization of science. This taxonomy is created from a list of entities, which \
+        represent Wikipedia topics and which should represent relevant concepts in academia. These entities \
+        are clustered into topics given a co-occurrence matrix. \n\n \
+        The problem is that some of these entities may instead represent modern day countries, regions, or cities. \
+        If possible, also identify institutions and films. \
+        \n\n \
+        Your task is to identify these entities and remove them from the taxonomy. \
+        I will give you a list of entities and you will return the list of entities to remove. \
+        \n\n \
+        Your response should be a single Python list of entities that should be removed. \
+        For example, if you think that the entities 'United States', 'United Kingdom', and 'Canada' \
+        should be removed, you would respond with ['United States', 'United Kingdom', 'Canada']. \
+        \n\n \
+        If you think that no entities should be removed, you would respond with an empty list, \
+        like this: []. \n\n \
+        Here is a first list of entities: \
+        \n\n \
+        {entities[:50]} \n\n"
+
+    response = bot.ask(initial_script)
+    logger.info(f"Initial response: {response}")
+
+    time.sleep(np.random.uniform(5, 20))
+    bot.ask(
+        "I will proceed to give you lists of entities, and you will return the list of entities to remove."
+    )
+
+    # Iteratively remove entities and prompt chatGPT
+    failed_attempts = 0
+    while len(entities) > 0:
+        if len(entities) >= 100:
+            # Get a sample of entities
+            entity_sample = random.sample(entities, 100)
+        else:
+            entity_sample = entities
+
+        # Define script & get response
```

Comment:
similarly to above, would probably be useful to have the script steps in a yaml file